### PR TITLE
feat: add 5-infra envs

### PR DESCRIPTION
**Pull request by rutalreja-deloitte**

### CODE REVIEW

File: 5-infrastructure/business_unit_1/development/development.auto.example.tfvars [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+parent_folder              = "000000000000"
+boa_gke_project_id         = "prj-bu1-z-boa-gke-xxxx" # 'z' is replaced by {d|n|p} according to environment {development|non-production|production}
+boa_ops_project_id         = "prj-bu1-z-boa-ops-xxxx" # 'z' is replaced by {d|n|p} according to environment {development|non-production|production}
+boa_sec_project_id         = "prj-bu1-z-boa-sec-xxxx" # 'z' is replaced by {d|n|p} according to environment {development|non-production|production}
+boa_sql_project_id         = "prj-bu1-z-boa-sql-xxxx" # 'z' is replaced by {d|n|p} according to environment {development|non-production|production}
+gcp_shared_vpc_project_id  = "prj-z-shared-base-xxxx" # 'z' is replaced by {d|n|p} according to environment {development|non-production|production}
+shared_vpc_name            = "vpc-yyy-shared-base"    # 'yyy' is replaced by {dev|npd|prd} according to environment {development|non-production|production}
+terraform_service_account  = "boa-terraform-z-sa@xxxxxxxxxx.iam.gserviceaccount.com"
+gke_cluster_1_cidr_block   = "000.000.000.000/00"
```
Comment (bharathkkb) [inline 5-infrastructure/business_unit_1/development/development.auto.example.tfvars:25]: Lets only get the project id's and SA email. I think everything else can be defaulted from 3-networks? Users can override this via variable if they change our instructions for 3-networks.

File: 5-infrastructure/business_unit_1/development/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+output "sql_2_ip_address" {
+  description = "PostgreSQL Instance 2 Private Ip"
+  value       = module.env.sql_outputs.sql2["Private IP Address"]
+}
+
+/******************************************
+  KMS Outputs
+*****************************************/
+
+output "kms_sa" {
```
Comment (bharathkkb) [inline 5-infrastructure/business_unit_1/development/outputs.tf:45]: Do we use the kms downstream?

File: 5-infrastructure/business_unit_1/development/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```
Comment (bharathkkb) [inline 5-infrastructure/business_unit_1/development/README.md:1]: We should add a root readme in 5-infrastructure that explains how to deploy this. You can use https://github.com/amandakarina/terraform-example-foundation/tree/feature/adds-step-5/5-app-infra as a starter

File: 5-infrastructure/business_unit_1/development/variables.tf [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+variable "location_primary" {
+  type        = string
+  description = "The primary region for deployment, if not set default locations for each resource are taken from variables file"
+  default     = "us-east1"
+}
+
+variable "location_secondary" {
+  type        = string
+  description = "The secondary region for deployment, if not set default locations for each resource are taken from variables file"
```
Comment (bharathkkb) [inline 5-infrastructure/business_unit_1/development/variables.tf:35]: nit: remove `if not set default locations for each resource are taken from variables file`

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR @rutalreja-deloitte

File: 5-infrastructure/business_unit_1/development/main.tf [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module "env" {
+  source = "../../modules/base-env"
+  env    = "dev"
+
+  parent_folder              = var.parent_folder
```
[the remark this answers is not in the document]
Comment (rutalreja-deloitte) [reply 5-infrastructure/business_unit_1/development/main.tf:21]: i'll take it out, it was there when base env was doing auto_project_id

File: 5-infrastructure/business_unit_1/development/outputs.tf (hunk 2 above)
Comment (rutalreja-deloitte) [reply 5-infrastructure/business_unit_1/development/outputs.tf:45]: no can take it out

File: 5-infrastructure/business_unit_1/development/development.auto.example.tfvars (hunk 1 above)
Comment (rutalreja-deloitte) [reply 5-infrastructure/business_unit_1/development/development.auto.example.tfvars:25]: shared_vpc_name should be taken as well imo, can be vpc-n-shared-base or vpc-npd-shared-base (Chris) or 
vpc-n-shared-base-spoke (hub&spoke flag mode)

**Review by bharathkkb — APPROVED** [review]

File: 5-infrastructure/business_unit_1/development/development.auto.example.tfvars (hunk 1 above)
Comment (bharathkkb) [reply 5-infrastructure/business_unit_1/development/development.auto.example.tfvars]: we should default to hub and spoke name, users can override if they want

**Pull request merged**